### PR TITLE
create fluent-bit-plugin-loki package

### PR DESCRIPTION
**Pull request by sanghanan**
Fixes:

Related: https://github.com/chainguard-dev/image-requests/issues/1659

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: fluent-bit-plugin-loki.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      repository: https://github.com/grafana/loki
+      tag: v${{package.version}}
+
+  - uses: go/bump
+    with:
+      deps: github.com/docker/docker@v25.0.5 golang.org/x/net@v0.23.0
+      replaces: go.etcd.io/bbolt=go.etcd.io/bbolt@v1.3.6
+
+  - runs: |
+      make fluent-bit-plugin
```
Comment (mamccorm) [inline fluent-bit-plugin-loki.yaml:32]: Could we use our go/build melange wrapper here instead? I had a look at the [Makefile](https://github.com/grafana/loki/blob/00bdd2f5b703991b280317ceff0fcf2eed1847d9/Makefile#L508) and it doesn't look like they are doing anything overly special.

go/build has optimisations built in for the binaries, so we should always try and use it where possible

Comment (sanghanan) [reply fluent-bit-plugin-loki.yaml]: Thanks for telling me! I've changed it to use go/build now.

File: fluent-bit-plugin-loki.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      expected-commit: b4f7181c7aa9484e66976e8a933111a9b85ea8c2
+      repository: https://github.com/grafana/loki
+      tag: v${{package.version}}
+
+  - uses: go/bump
+    with:
+      deps: github.com/docker/docker@v25.0.5 golang.org/x/net@v0.23.0
+      replaces: go.etcd.io/bbolt=go.etcd.io/bbolt@v1.3.6
+  
+  - uses: go/build
```
Comment (sanghanan) [inline fluent-bit-plugin-loki.yaml:31]: Trying to replicate this using go/build, but running into segmentation error
```
go build -ldflags "\
    -s -w \
    -X github.com/grafana/loki/v3/pkg/util/build.Branch=$(git rev-parse --abbrev-ref HEAD) \
    -X github.com/grafana/loki/v3/pkg/util/build.Version=3.0.0 \
    -X github.com/grafana/loki/v3/pkg/util/build.Revision=$(git rev-parse --short HEAD) \
    -X github.com/grafana/loki/v3/pkg/util/build.BuildUser=$(whoami)@$(hostname) \
    -X github.com/grafana/loki/v3/pkg/util/build.BuildDate=$(date -u +"%Y-%m-%dT%H:%M:%SZ")" \
    -tags netgo \
    -buildmode=c-shared \
    -o clients/cmd/fluent-bit/out_grafana_loki.so \
    ./clients/cmd/fluent-bit/
    ```

Comment (jonjohnsonjr) [reply fluent-bit-plugin-loki.yaml:31]: Sent out a change to melange to plumb the `-buildmode` flag through `go/build`: https://github.com/chainguard-dev/melange/pull/1210

Once that's in (and we bump our deps here) this diff should do the trick:

```
diff --git a/fluent-bit-plugin-loki.yaml b/fluent-bit-plugin-loki.yaml
index f847f89b1..0f312d2a2 100644
--- a/fluent-bit-plugin-loki.yaml
+++ b/fluent-bit-plugin-loki.yaml
@@ -33,6 +33,7 @@ pipeline:
       packages: ./clients/cmd/fluent-bit/
       modroot: .
       output: out_grafana_loki.so
+      buildmode: c-shared
       ldflags: |
         -s -w
         -X github.com/grafana/loki/v3/pkg/util/build.Branch=$(git rev-parse --abbrev-ref HEAD)
@@ -40,7 +41,6 @@ pipeline:
         -X github.com/grafana/loki/v3/pkg/util/build.Revision=$(git rev-parse HEAD)
         -X github.com/grafana/loki/v3/pkg/util/build.BuildUser=$(whoami)@$HOSTNAME
         -X github.com/grafana/loki/v3/pkg/util/build.BuildDate=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
-        -buildmode=c-shared
       tags: netgo

   - runs: |
```

File: fluent-bit-plugin-loki.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    with:
+      packages: ./clients/cmd/fluent-bit/
+      modroot: .
+      output: out_grafana_loki.so
```
Comment (jonjohnsonjr) [inline fluent-bit-plugin-loki.yaml:35]:
```suggestion
      output: out_grafana_loki.so
      buildmode: c-shared
```

File: fluent-bit-plugin-loki.yaml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      ldflags: |
+        -s -w
+        -X github.com/grafana/loki/v3/pkg/util/build.Branch=$(git rev-parse --abbrev-ref HEAD)
+        -X github.com/grafana/loki/v3/pkg/util/build.Version=${{package.version}}
+        -X github.com/grafana/loki/v3/pkg/util/build.Revision=$(git rev-parse HEAD)
+        -X github.com/grafana/loki/v3/pkg/util/build.BuildUser=$(whoami)@$HOSTNAME
+        -X github.com/grafana/loki/v3/pkg/util/build.BuildDate=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+        -buildmode=c-shared
```
Comment (jonjohnsonjr) [inline fluent-bit-plugin-loki.yaml:43]:
```suggestion
```

File: fluent-bit-plugin-loki.yaml (hunk 3 above)
Comment (jonjohnsonjr) [reply fluent-bit-plugin-loki.yaml:35]: You might need to rebase onto main to get this to work.

**Review by jamonation — APPROVED** [review]
Looking good, thanks for working through this one

**Pull request merged**